xray/plot/plot.py

```diff
@@ -492,13 +364,29 @@ def _plot2d(plotfunc):
     plotfunc.__doc__ = '\n'.join((plotfunc.__doc__, commondoc))
 
     @functools.wraps(plotfunc)
-    def newplotfunc(darray, x=None, y=None, ax=None, xincrease=None, yincrease=None,
-                    add_colorbar=True, add_labels=True, vmin=None, vmax=None, cmap=None,
-                    center=None, robust=False, extend=None, levels=None, colors=None,
-                    **kwargs):
+    def newplotfunc(darray, x=None, y=None, ax=None, row=None, col=None,
+                    col_wrap=None, xincrease=None, yincrease=None,
+                    add_colorbar=True, add_labels=True, vmin=None, vmax=None,
+                    cmap=None, center=None, robust=False, extend=None,
+                    levels=None, colors=None, **kwargs):
         # All 2d plots in xray share this function signature.
         # Method signature below should be consistent.
 
+        # Handle facetgrids first
+        if row or col:
+            allargs = locals().copy()
+            allargs.update(kwargs)
+            del allargs['kwargs']
+
+            # Deleting allows us to use better FacetGrid defaults
```

Comment:
Another option here would be to switch these defaults to `None`. In this case, we could just pick the right default for the situation.